profiler/profile.go

```diff
@@ -44,48 +44,113 @@ const (
 	MetricsProfile
 )
 
-func (t ProfileType) String() string {
-	switch t {
-	case HeapProfile:
-		return "heap"
-	case CPUProfile:
-		return "cpu"
-	case MutexProfile:
-		return "mutex"
-	case BlockProfile:
-		return "block"
-	case GoroutineProfile:
-		return "goroutine"
-	case expGoroutineWaitProfile:
-		return "goroutinewait"
-	case MetricsProfile:
-		return "metrics"
-	default:
-		return "unknown"
+type profileType struct {
```

Comment:
Perhaps a better name is `collector` (in the context of the package `profiler.collector` kind of makes sense)

Reply:
Yeah, that might be a better name. I'll try it.

Reply:
3811119de2c00913c525075de16d0297c2a3cf07

Reply:
What made you go back here? You can pick whatever but I think `profileType` is not a very good name for this struct.

Reply:
I kept coming back to this code and couldn't warm up to the name `collector`. To me this struct is holding the implementation details of a profile type, and calling it by any other name feels odd to me. Considering that this is a private type inside the profiler pkg, I'd like to stick with `profileType`. Let me know if that's ok on your end or if you have more ideas on alternative names here.

Reply:
I see. Since you basically own this code now, I want you to be first and foremost comfortable with the naming we choose. There are two things I didn't like about this name:
* The suffix `Type`, which seems odd for any type
* The fact that we already have a `ProfileType` type
* Anything ending with "type" sounds more like an enum than a struct.

I'm not sure what else to suggest to be honest. Naming is hard. Perhaps adding yet another word? Something like `profileTypeRequest`? Or, you could consider calling this `profile` and renaming the current struct with that name to `pprofFile`.

Then again, looking at the suggested options, perhaps what we currently have work. That struct does represent after all, a profile type, with it's various information such as name, way to retrieve it (collect func), etc.

Fine by me. I was mostly curious why you came back to this.

Reply:
Thanks for the explanations @gbbr. I don't see the issues you raised as problematic and think having the implementation of `ProfileType` named `profileType` is quite natural. That being said, at the end of the day this is mostly about personal preference on my end. So since I'm currently maintaining this pkg I'll go with a name that makes sense to me. If somebody else comes along later they can easily rename it. It's private type name after all. Hope that's okay : )

Reply:
Yep, agreed. Resolving.